### PR TITLE
P5AP-JA 140 frontend pages dashboard

### PR DESCRIPTION
**Pull request by jagadeepavula**
[PC-140](https://makeitmvp.atlassian.net/browse/P5AP-140)

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring

## Description

- Changes made?
created new component, challengePage and copies challengeCard component from different branch
- Reason for changes?
needed new component as per figma designs
- Behavior before?
No challenge page before
- Behavior after?
Has the challenge page as per figma designs with cards and filter
- How did you test?
Locally tested by running it in testpage and verified the designs are the same as per figma

## Checklist

- [ ] I have performed a self-review of my code.
- [x ] I have manually tested the new feature and found it to be working as expected.
- [ x] All commits in this PR are related to the issue.
- [ x] Base branch for this PR is set to `develop`.
- [ x] JIRA ticket # and my initials are noted in this PR's name (format example: #57-EB Add Pull Request template)

**Reminder**: every PR needs at least 1 peer approval before merging to the `develop` branch. Once a main ticket has been completed, along with all subtasks, associated code can be merged to `main`.

### CODE REVIEW

**Review by ToluAjisola — APPROVED** [review]
Just making a note about moving the styling to the scss file so the jsx file is easier to read, later In the challengecard.jsx

**Comment by jagadeepavula** [comment]
I have created a separate file for styling now and copied all necessary styling code there. I tested in my local and it works great and is ready for re-review.

**Review by ToluAjisola — CHANGES_REQUESTED** [review]
merge conflict

**Comment by jagadeepavula** [comment]
I have resolved the merge conflicts.

File: src/components/ChallengePage/ChallengePage.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          {selectedDifficulties.map((difficulty) => (
+            <span key={difficulty} className="filter-tab">
+              {difficulty}
+            </span>
+          ))}
+        </div>
+
+        <div className="card-container">
+          {filteredChallenges.map((challenge, index) => (
+            <ChallengeCard key={index} challenge={challenge} />
```
Comment (alinix1) [inline src/components/ChallengePage/ChallengePage.jsx]: Might not be best practice to use the index as a key but find a unique identifier or something else instead.

**Review by ToluAjisola — APPROVED** [review]

File: src/components/ChallengePage/ChallengePage.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { mockChallenges } from "../../mock-data/mockChallenges";
+import ChallengeCard from "../ChallengeCard/ChallengeCard";
+import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
+import filterIcon from "../../assets/images/filterIcon.svg";
+import "./ChallengePage.scss";
+
+const ChallengePage = () => {
+  const [selectedCategories, setSelectedCategories] = useState([
```
Comment (robester0403) [inline src/components/ChallengePage/ChallengePage.jsx]: I think you just need an array, not a state. I don't see you changing this array anywhere

File: src/components/ChallengePage/ChallengePage.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    "Development",
+    "Python",
+    "React",
+    "Java",
+    "DSA",
+  ]);
+  const [selectedDifficulties, setSelectedDifficulties] = useState([
```
Comment (robester0403) [inline src/components/ChallengePage/ChallengePage.jsx]: Same here

File: src/components/ChallengePage/ChallengePage.jsx (hunk 1 above)
Comment (robester0403) [reply src/components/ChallengePage/ChallengePage.jsx]: if there is an unique ID then use it. Maybe even combine the challenge name with the index to make it unique. 

Index is ok if it's some list that doesn't change much in an isolated case. Like some small component.

**Comment by jagadeepavula** [comment]
I have changed the useState to an array and also used category and task in place of index for a unique ID.

**Review by robester0403 — APPROVED** [review]

**Pull request merged**